CONTRIBUTING.md

```diff
@@ -88,11 +94,183 @@ We recommend the following commit structure in the following order:
+
+On Windows we have also found the [Python Launcher for Windows](https://docs.python.org/3/using/windows.html) to be a reasonable alternative.
+
+After you have set up your Python 3 installation, install tox (the `--user` flag is important **never use `sudo` to install Python packages!**)
+
+```bash
+python3 -m pip install --user tox
+```
+
+Check that tox is in the binary path
+
+```bash
+tox --version
+```
+
+If not, find the user base binary directory
+
+```bash
+python -m site --user-base
+#~ /Users/<username>/Library/Python/3.7
+```
+
+Append `bin` to the directory returned and add this to your path by updating `~/.profile`. For example you might add the following:
+
+```bash
+export PATH=~/Library/Python/3.7/bin/:$PATH
+```
+
+## Setup Development Environment
+
+Clone the `mbed-tools` GitHub repository.
+
+```bash
+git clone git@github.com:ARMmbed/mbed-tools.git
+```
+
+Set up the development environment using tox (tox will create a development environment at mbed-tools/.venv):
+
+```bash
+cd mbed-tools/
+tox -e dev
+source .venv/bin/activate
+```
+
+## Unit Tests, Code Formatting and Static Analysis
+
+After you have activated your development environment, run `tox -e py37` to run
+unit tests and static code analysis checks:
+
+```bash
+tox -e py37
+```
+
+This will run `black`, `flake8`, `mypy` and `pytest`. If you would like to run these tools individually, see below:
+
+Run unit tests:
+
+```bash
+pytest
+```
+
+Run code formatter (it will format files in place):
+
+```bash
+black .
+```
+
+Run static analysis (note that no output means all is well):
+
+```bash
+flake8
+```
+
+Perform static type check:
+
+```bash
+mypy src/mbed_tools
+```
+
+## Documenting code
+
+Inclusion of docstrings is needed in all areas of the code for Flake8 checks in the CI to pass.
+
+We use [google-style](http://google.github.io/styleguide/pyguide.html#381-docstrings) docstrings.
+
+To set up google-style docstring prompts in Pycharm:
+
+* in the menu navigate to Preferences > Tools > Python Integrated Tools
+* in the dropdown for docstring format select 'Google'
+
+For longer explanations, you can also include markdown. Markdown can also be kept in separate files in the
+`docs/user_docs` folder and included in a docstring in the relevant place using the [reST include](https://docutils.sourceforge.io/docs/ref/rst/directives.html#including-an-external-document-fragment) as follows:
+
+```python
+    .. include:: ../docs/user_docs/documentation.md
+```
+
+### Building docs locally
+
+You can do a preview build of the documentation locally by running:
+
+```bash
+generate-docs
+```
+
+This will generate the docs and output them to `local_docs`.
+This should only be a preview. Since documentation is automatically generated
+by the CI you shouldn't commit any docs html files manually. The docs will be
+generated and checked in as part of the release process.
+
+### Viewing docs generated by the CI
```

Comment:
Does a contributor need to care about building docs locally or viewing them in the CI? Is this just noise to a new developer on the project? I think "Type hints" are definitely more relevant to the casual contributor than viewing the generated docs. Maybe this section should move below the "Code Climate" section, if we need it at all?

Reply:
I could foresee PRs coming that edit these files not knowing they are autogenerated. I think it's good to have a section on this, but I'll move it.